dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -312,11 +316,9 @@ public void scan() throws CDIException {
 
             //and pull out the corresponding classes
             for (String className : rawBeanclassNames) {
-                Class<?> clazz = this.allClasses.get(className);
-                if (clazz != null) {
-                    if (clazz.getClassLoader() == classLoader || !isAccessibleBean(clazz)) {
```

Comment:
I think you still need this bit. Even if you can replace the classloader check using the shadow service, you still need to check if the class already exists in child BDAs (`isAccessibleBean`)